services/consensuscontext/ordered_committee.go

```diff
@@ -108,21 +104,25 @@ func (s *service) callElectionsSystemContract(ctx context.Context, blockHeight p
 		return nil, errors.Errorf("call system %s.%s returned corrupt output value", systemContractName, systemMethodName)
 	}
 	joinedAddresses := arg0.BytesValue()
+	return splitAddresses(joinedAddresses), nil
+}
 
+func splitAddresses(joinedAddresses []byte) []primitives.NodeAddress {
 	numAddresses := len(joinedAddresses) / digest.NODE_ADDRESS_SIZE_BYTES
 	res := make([]primitives.NodeAddress, numAddresses)
 	for i := 0; i < numAddresses; i++ {
-		res[i] = joinedAddresses[20*i : 20*(i+1)]
+		res[i] = joinedAddresses[digest.NODE_ADDRESS_SIZE_BYTES*i : digest.NODE_ADDRESS_SIZE_BYTES*(i+1)]
 	}
-	return res, nil
+	return res
 }
 
-func toNodeAddresses(nodes map[string]config.ValidatorNode) []primitives.NodeAddress {
-	nodeAddresses := make([]primitives.NodeAddress, len(nodes))
-	i := 0
+func generateGenesisCommittee(nodes map[string]config.ValidatorNode) []primitives.NodeAddress {
+	res := make([]primitives.NodeAddress, 0, len(nodes))
 	for _, value := range nodes {
```

Comment:
range will diverge on different nodes?
We need to inflict order on genesis committee list (based on height) - same as before - support the blocks until election

Reply:
after copy there is sort so no issue about that. but it doesn't do random here. maybe we do need a version of orderCommitteeWithInput to be exposed ?